mapflow/mapflow.py

```diff
@@ -2187,15 +2187,25 @@ def preview_xyz(self, provider, image_id):
             for l in self.iface.mapCanvas().layers():
                 if l.name() == 'OpenStreetMap':
                     QgsProject.instance().removeMapLayers([l.id()])  
-            self.alert(self.tr("Preview is unavailable for the provider {}. \nOSM layer will be added instead.").format(provider.name))
+            self.alert(self.tr("Preview is unavailable for the provider {}. \nOSM layer will be added instead.").format(provider.name), QMessageBox.Information)
             # Add OSM instaed of preview, if it is unavailable (for Mapbox)
             osm = 'type=xyz&url=https://tile.openstreetmap.org/{z}/{x}/{y}.png&zmax=19&zmin=0'
             layer = QgsRasterLayer(osm, 'OpenStreetMap', 'wms')  
-            self.result_loader.add_layer_to_bottom(layer)
+            for l in self.iface.mapCanvas().layers(): 
+                if l.name() == 'OpenStreetMap' or l.name() == 'ArcGIS World Imagery':
```

Comment:
Comparison by name is far from perfect.
1. User can rename the layer
2. User can have his own providers (see "settings" tab, there are controls to add one)

We can do it by accumullating the plugin-added basemap layers in a list and check for the layers' positions on addition of a new layer. However, we should track them and remove the ones that are removed by user.